src/java/org/jivesoftware/webchat/ChatSession.java

```diff
@@ -440,10 +474,15 @@ private void checkForEmptyRoom() {
         // See if we are the last one in the chat room.
         if (groupChat.getOccupantsCount() == 1) {
             // Leave the room.
-            groupChat.leave();
+            try {
+				groupChat.leave();
+			} catch (NotConnectedException e) {
+				// TODO Auto-generated catch block

```

Comment:
Please don't ever do ``e.printStackTrace()``. Write the exception to the log instead.